File: src/main/java/net/tassia/event/EventHandler.java
package net.tassia.event;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Annotate a method with this annotation to register it as an event listener.
 * @since EventAPI 1.0
 * @author Tassilo
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface EventHandler {
}

File: src/main/kotlin/net/tassia/event/EventHandler.kt
package net.tassia.event

/**
 * Annotate a method with this annotation to register it as an event listener.
 *
 * @since EventAPI 1.0
 * @author Tassilo
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION)
annotation class EventHandler
